indexer/src/main.rs

```diff
@@ -0,0 +1,186 @@
+                    pool,
+                    &shard.state_changes,
+                    streamer_message.block.header.height,
+                )
+            });
+
+            try_join_all(futures).await.map(|_| ())
+        };
+
+        // StateChange related to Account
+        let account_changes_future = adapters::account_changes::store_account_changes(
+            pool,
+            &streamer_message.shards,
+            &streamer_message.block.header.hash,
+            streamer_message.block.header.timestamp,
+        );
+
+        try_join!(
+            execution_outcomes_future,
+            accounts_future,
+            access_keys_future,
+            assets_events_future,
+            account_changes_future,
+        )?;
+    } else {
+        try_join!(
+            execution_outcomes_future,
+            accounts_future,
+            assets_events_future
+        )?;
+    }
+
+    Ok(())
+}
+
+#[actix::main]
+async fn main() -> anyhow::Result<()> {
+    // We use it to automatically search the for root certificates to perform HTTPS calls
+    // (sending telemetry and downloading genesis)
+    openssl_probe::init_ssl_cert_env_vars();
+
+    dotenv::dotenv().ok();
+
+    let opts: Opts = Opts::parse();
+
+    configs::init_tracing(opts.debug)?;
+
+    // We establish connection as early as possible as an additional sanity check.
+    // Indexer should fail if .env file with credentials is missing/wrong
+    let pool = models::establish_connection(&opts.database_url);
+
+    let strict_mode = !opts.non_strict_mode;
+
+    // We want to prevent unnecessary SELECT queries to the database to find
+    // the Transaction hash for the Receipt.
+    // Later we need to find the Receipt which is a parent to underlying Receipts.
+    // Receipt ID will of the child will be stored as key and parent Transaction hash/Receipt ID
+    // will be stored as a value
+    let receipts_cache: receipts_cache::ReceiptsCache =
+        std::sync::Arc::new(Mutex::new(SizedCache::with_size(100_000)));
+
+    let config: near_lake_framework::LakeConfig = opts.to_lake_config().await;
+    let (_, stream) = near_lake_framework::streamer(config);
+
+    tokio::spawn(async move {
+        tracing::info!(
+            target: INDEXER_FOR_EXPLORER,
+            "Starting Indexer for Explorer (lake)...",
+        );
+        let mut handlers = tokio_stream::wrappers::ReceiverStream::new(stream)
+            .map(|streamer_message| {
+                info!(
+                    target: crate::INDEXER_FOR_EXPLORER,
+                    "Block height {}", &streamer_message.block.header.height
+                );
+                handle_message(
+                    &pool,
+                    streamer_message,
+                    strict_mode,
+                    std::sync::Arc::clone(&receipts_cache),
+                )
+            })
+            .buffer_unordered(1usize);
+
+        while let Some(_handle_message) = handlers.next().await {}
```

Comment:
Major: we will silently die without even printing the error to logs

Reply:
Added error handling 👍🏽. Decided to kill the entire process as that seems to be the consensus in https://github.com/near/near-indexer-events/pull/32#discussion_r1042698543